applications/StructuralMechanicsApplication/custom_elements/timoshenko_beam_element_2D2N.cpp

```diff
@@ -0,0 +1,1051 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:  Alejandro Cornejo
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_elements/timoshenko_beam_element_2D2N.h"
+#include "custom_utilities/structural_mechanics_element_utilities.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+void TimoshenkoBeamElement2D2N::Initialize(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Initialization should not be done again in a restart!
+    if (!rCurrentProcessInfo[IS_RESTARTED]) {
+        if (this->UseGeometryIntegrationMethod()) {
+            if (GetProperties().Has(INTEGRATION_ORDER) ) {
+                const SizeType integration_order = GetProperties()[INTEGRATION_ORDER];
```

Comment:
i would say the enum is numbered. are u sure it is not enough to do

~~~cpp
mThisIntegrationOrder=2 (say 2)
~~~